app/models/board.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 class Board < ApplicationRecord
-  has_many :discussions
+  has_many :discussions, dependent: :destroy
```

Comment:
If a board gets removed, it might be better to use dependent: :nullify so that the discussion threads are preserved.